### PR TITLE
daemon/containerd: Replace pkg/errors with std errors

### PR DESCRIPTION
**Pull request by vvoland**
**- What I did**
Replaced `errors.Wrap` with `fmt.Errorf`

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by rumpl — APPROVED** [review]

**Review by thaJeztah — CHANGES_REQUESTED** [review]
leaving a quick "request changes"; I'll add more details below

**Comment by thaJeztah** [comment]
From a conversation on slack;

We need to decide which direction we want to go with errors, as there were multiple PRs / discussions related to that. On one hand, it'd be great if we could just depend on `stdlib`; it's what the language provides, and what most packages we use will be using.

On the other hand, `stdlib` errors may be limiting in some areas, and we must look if we can live with that :smile:

- stdlib now provides `errors.Join`, which allows for "multi-errors" (w00t!)
- current versions of go also allow _multiple_ errors to be included in an error-format string, so this is now allowed:
 
    ```go
    fmt.Errorf("foo: %w, and other errors: %w, %w", err, err1, err2")
    ```

  (I must admit that I did not yet use that format, and don't know how those nested errors are unwrapped; possibly that change relates to the "multi-errors")
- however, stdlib error's formatting of multi-errors is not "ideal", which led us to PR's like https://github.com/moby/moby/pull/46099, https://github.com/moby/moby/pull/46188, and https://github.com/moby/moby/pull/46275

For `github.com/pkg/errors`:

- the package has been archived by the author (but has been stable, so that alone should not necessarily have to be a concern)
- it allows for traces to be obtained from errors (although I'm not sure if that's always provided); this allows us to do things like https://github.com/moby/moby/pull/43252, which _COULD_ be quite useful. Is something similar possible with `stdlib` errors? (to be looked into)
- similar / related, I know @tonistiigi was able to get a full trace of errors as part of OTEL traces. This was possible even up to containerd internals. The same was not possible with stdlib errors, and containerd removed uses of `pkg/errors`, so perhaps that ship has sailed (see https://github.com/containerd/containerd/pull/6937#issuecomment-1125066122)

In short (I'll open a tracking ticket / discussion for this), we should decide:

- what we want to get out of errors
- do a more in-depth comparison between implementations to have better faces about what's possible (and what not) which each.
- decide what implementation to use going forward; I'd be happy if that is "stdlib", but we should decide (and write down) that decision, and base it on actual needs / requirements (beyond use stdlib, not external dependencies)
- if that decision leads us to using `pkg/errors`, we should try to reach out to the original author to see if they're open to handing over maintenance, or "hard-fork" it, and maintain our own; see https://github.com/moby/buildkit/pull/3305#issuecomment-1336203192

**Comment by thaJeztah** [comment]
- I also captured the above in a discussion: https://github.com/moby/moby/discussions/46358
